portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,10 +34,64 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  private final static Logger LOGGER = Logger.getLogger(DataServlet.class.getName());
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    // Get all the comments from the database
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<String> comments = new ArrayList<String>();
+    for (Entity entity : results.asIterable()) {
+      String comment = (String) entity.getProperty("comment");
+      comments.add(comment);
+    }
+
+    response.setContentType("application/json");
+    response.getWriter().println(toJSONString(comments, "comments"));
+  }
+
+  /*
+   * Helper function to store an array in JSON format
+   */
+  private String toJSONString(List<String> array, String arrayName) {
+    Gson gson = new Gson();
+    if (array != null) {
+      String json = "{ \"" + arrayName + "\" :"
+          +  gson.toJson(array).toString() + "}";
+          System.out.println(json);
+      return json;
+    }
+    return null;
+
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String feedback = request.getParameter("text-input");
+    long timestamp = System.currentTimeMillis();
+
+    // Only add feedback if valid input
+    if (feedback == "") {
+      LOGGER.warning("No input");
+      return;
+    }
+
+    else {
```

Comment:
Nit: [no break after a closing brace if it is followed by an else](https://google.github.io/styleguide/javaguide.html#s4.1.2-blocks-k-r-style). Clang might be able to tidy this up for you.

Reply:
Clang didn't work, but did it manually